Updates dependencies to latest

Wackymax opened the pull request.

Wackymax committed:
Updated dependencies

Wackymax committed:
Add changelog

vkammerer commented:
@bahung1221  Is there anything that prevents this PR from being merged?

The upgrade to `protobuf: ^3.0.0` is critical in our case as we need it in our app.

bahung1221 committed:
Update pubspec.yaml

Bump minimum Dart SDK version to support protobuf 3.0.0

Because vector_tile depends on protobuf >=3.0.0 which requires SDK version >=2.19.0 <4.0.0, version solving failed.

bahung1221 commented:
Previously I tested it on my old laptop which old Dart SDK and it failed to resolve dependencies, then I just forgot it completely, sorry for that 😄 

Updated the minimum Dart SDK to 2.19.0 for compatibility with protobuf 3.0.0

![Uploading Screenshot 2023-12-02 at 12.01.38.png…]()

bahung1221 commented:
published v2.0.0, thanks a lot!
